### PR TITLE
upstream sync 180613

### PR DESCRIPTION
**Pull request by whchung**
Keep in sync with TensorFlow upstream up to 06/13/2018

### CODE REVIEW

**Comment by whchung** [comment]
@iotamudelta please help review change in this PR, in particular commit # 29e3027 where i tried to follow your steps to make `bias_op` now be `rocPRIM`-aware.

File: tensorflow/core/kernels/bias_op_gpu.cu.cc [hunk 1]
```diff
@@ -24,6 +24,14 @@ limitations under the License.
...
 #include "tensorflow/core/util/gpu_kernel_helper.h"
 
+#include "tensorflow/core/framework/tensor.h"
+#include "tensorflow/core/framework/tensor_shape.h"
+#include "tensorflow/core/framework/tensor_types.h"
+#include "tensorflow/core/platform/types.h"
+
+#include "tensorflow/core/kernels/reduction_gpu_kernels.cu.h"
+#include "tensorflow/core/kernels/reduction_ops_common.h"
+
```
Comment (iotamudelta) [inline tensorflow/core/kernels/bias_op_gpu.cu.cc]: you may need the

```
#if GOOGLE_CUDA
namespace gpuprim = ::cub;
#elif TENSORFLOW_USE_ROCM
namespace gpuprim = ::hipcub;
#endif
```

definition here

Comment (iotamudelta) [reply tensorflow/core/kernels/bias_op_gpu.cu.cc]: although, probably reduction_gpu pulls that in, on second thought

Comment (whchung) [reply tensorflow/core/kernels/bias_op_gpu.cu.cc]: yeah that's what i found

**Comment by whchung** [comment]
@parallelo it seems one of CI node is idling, while the other one is busy checking #25 . could you help check the configuration and see if we can improve the node utilization?

**Comment by parallelo** [comment]
Yes, I noticed that too.  We had two Jenkins workers going concurrently yesterday.  I've started a Jenkins "safeRestart" to see if that clears up the issue.  Will keep my eye on it.

**Comment by parallelo** [comment]
Jenkins came back from the restart recently, and now a job is running on the previously always-idle worker; so I believe the issue is resolved.  Let's keep watch for any future idle workers.

**Comment by whchung** [comment]
5 additional failures. I'll triage them after finishing up HIP PR #457

**Pull request merged**